Update gpfdist.c

ddanese opened the pull request:
When compiling gpfdist on ubuntu, the compile fails because of the compiler flag to treat all warnings as errors.
You get a warning when calling freopen and not capturing the result of the call.
This change is simple gpfdist successfully compiles on Ubuntu 15.10 with the change.

ddanese commented:
This is a redirect of stderr and stdout, so checking for errors is not a bad idea.  freopen returns null if an error occured, what would be the proper response that in code? Do you just want to return a negative value or does something else need to happen to alert the program? I see that on success it returns 0, so if that is following the somewhat standard, would -1 suffice?

yydzero commented:
maybe print a message when function call failed.

On Thu, Jan 21, 2016 at 8:46 AM, Dan Danese <notifications@github.com>
wrote:

> This is a redirect of stderr and stdout, so checking for errors is not a
> bad idea. freopen returns null if an error occured, what would be the
> proper response that in code? Do you just want to return a negative value
> or does something else need to happen to alert the program? I see that on
> success it returns 0, so if that is following the somewhat standard, would
> -1 suffice?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/greenplum-db/gpdb/pull/289#issuecomment-173414783>.
>



-- 
Best Regards,
Yandong

ddanese commented:
Updated the pull request code. Added a check for null which indicates an error on opening the file (the redirect).  If the file handle is null, it will return -1.

Chibin commented:
We have similar changes for this portion that I have on my pull request. I'll probably wait for yours to get approved instead.
https://github.com/Chibin/gpdb/commit/bc2394f3f326ae0a3dc6b819b37ad414b603e379

hlinnaka commented:
Would be nice to also tell the user why it failed, with strerror().

ddanese commented:
Updated to add the strerror call.

yydzero commented:
LGTM

hlinnaka commented:
I was about to push this with some further nitpicking:

* reworded the message slightly (lower case, removed "Error:") to match the style of other errors in the file more closely. (not that all the existing messages are consistent anyway, though...)
* added a newline to the error message
* moved the variable declaration to beginning of block (as required in plain C, some compilers might complain about that. We have a few more instances of that elsewhere in the file, but let's be tidy when we can) 
* squashed all the commits into one, edited the commit message accordingly.

But before I push, I thought I'd better check with you that this still looks OK to you, @ddanese, since I wanted to keep your details in the Author field:

https://github.com/hlinnaka/gpdb/commit/a3ab4321d88ae8c61cd449bd304c78e3fbd58775

So are you OK with that?

ddanese commented:
Yep. Thanks.

hlinnaka commented:
Ok, pushed. Thanks!
